corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -72,17 +74,34 @@ hqDefine("geospatial/js/geospatial_map", [
                 }
             });
 
+            function changeCaseMarkerColor(selectedCase, newColor) {
+                const caseId = selectedCase.case_id;
+                const caseMarker = Object.entries(caseMarkers).find(([key]) => key === caseId);
+                if (caseMarker) {
+                    const [caseId, marker] = caseMarker;
+                    let element = marker.getElement();
+                    let svg = element.getElementsByTagName("svg")[0];
+                    let path = svg.getElementsByTagName("path")[0];
+                    path.setAttribute("fill", newColor);
+                } else {
+                    console.error("No marker was found to be associated with a case!");
+                }
+            };
+
             function filterCasesInPolygon(polygonGeometry) {
                 userFilteredCases = [];
                 cases.filter(function (currCase) {
                     var coordinates = [currCase.coordinates.lng, currCase.coordinates.lat];
                     var point = turf.point(coordinates);
                     var caseIsInsidePolygon = turf.booleanPointInPolygon(point, polygonGeometry);
                     if (caseIsInsidePolygon) {
-                        userFilteredCases.push(currCase)
-                    };
+                        userFilteredCases.push(currCase);
+                        changeCaseMarkerColor(currCase, selectedMarkerColor);
+                    } else {
+                        changeCaseMarkerColor(currCase, defaultMarkerColor)
+                    }
                 });
-                console.log(userFilteredCases);
+                console.log("Filtered Cases: ", userFilteredCases);
```

Comment:
@SmittieC Do we want this console.log here?

Reply:
Probably not. Will remove

Reply:
https://github.com/dimagi/commcare-hq/pull/33059/commits/0405303f07b0c161e79119ba7000699cdfb57345